hibernate-core/src/main/java/org/hibernate/boot/registry/classloading/internal/AggregatedServiceLoader.java

```diff
@@ -216,22 +141,16 @@ private Collection<S> loadAll() {
 				 * See https://hibernate.atlassian.net/browse/HHH-13551.
 				 */
 				while ( hasNextIgnoringServiceConfigurationError( providerIterator ) ) {
-					Supplier<S> provider = providerIterator.next();
+					ServiceLoader.Provider<S> provider = providerIterator.next();
 					collectServiceIfNotDuplicate( result, alreadyEncountered, provider );
 				}
 			}
 
 			return result;
 		}
 
-		@SuppressWarnings("unchecked")
-		private Stream<? extends Supplier<S>> providerStream(ServiceLoader<S> serviceLoader) {
-			try {
-				return ( (Stream<? extends Supplier<S>>) SERVICE_LOADER_STREAM_METHOD.invoke( serviceLoader ) );
-			}
-			catch (RuntimeException | IllegalAccessException | InvocationTargetException e) {
-				throw new AssertionFailure( "Error calling ServiceLoader.stream()", e );
-			}
+		private Stream<ServiceLoader.Provider<S>> providerStream(ServiceLoader<S> serviceLoader) {
+			return serviceLoader.stream();
 		}
```

Comment:
I don't see a reason not to inline this?

Reply:
right, thanks - clearly forgot